learningresources/api.py

```diff
@@ -0,0 +1,113 @@
+"""
+Helper functions for using the models, so external
+apps don't tie functionality to internal implementation.
+"""
+
+from datetime import datetime
+
+from django.db import transaction
+
+from learningresources.models import (
+    Course, Repository, LearningResource, LearningResourceType
+)
+
+TYPE_LOOKUP = {}
+
+
+def create_course(org, course_number, semester, user_id):
+    """
+    Add a course to the database.
+    Args:
+        org (unicode): organization
+        course_number (unicode): course number
+        semester (unicode): semester
+        user_id (int): primary key of user creating the course
+    Raises:
+        ValueError: Duplicate course
+    Returns: None
+    """
+    # Check on unique values before attempting a get_or_create, because
+    # items such as import_date will always make it non-unique.
+    unique = {
+        "org": org, "course_number": course_number, "semester": semester,
+        "repository_id": get_temp_repository(user_id).id
+    }
+    if Course.objects.filter(**unique).exists():
+        raise ValueError("Duplicate course")
+    kwargs = {
+        "org": org, "course_number": course_number, "semester": semester,
+        'imported_by_id': user_id,
+        "repository_id": get_temp_repository(user_id).id
+    }
+    with transaction.atomic():
+        course, _ = Course.objects.get_or_create(**kwargs)
+    return course
+
+
+def get_temp_repository(user_id):
+    """
+    Use this for now so we can import *something*.
+
+    Args:
+        user_id (int): primary key of user creating the course
+    Returns:
+        repository (learningresources.Repository): repository
+    """
+    repos = Repository.objects.all()
+    if repos.count() > 0:
+        return repos[0]
+    with transaction.atomic():
+        return Repository.objects.create(
+            name="demo",
+            create_date=datetime.now(),
+            created_by_id=user_id,
+        )
+
+
+def create_resource(kwargs):
+    """
+    Create a learning resource.
+
+    Args:
+        course (learningresources.Course): course
+        parent (learningresources.LearningResource): parent LearningResource
+        resource_type (unicode): name of LearningResourceType
+        title (unicode): title of resource
+        content_xml (unicode): XML
+        mpath (unicode): materialized path
```

Comment:
I did that intentionally to keep the line length down. I don't mind if you prefer to change it.